src/client/components/ListOfWeeksWithDetail/index.js

```diff
@@ -0,0 +1,127 @@
+import Alert from "react-bootstrap/Alert";
+import PropTypes from "prop-types";
+import React from "react";
+import { Trans, useTranslation } from "react-i18next";
+import { userDataPropType } from "../../commonPropTypes";
+import { startAndEndDate, toWeekString } from "../../../utils/retroCertsWeeks";
+import programPlan from "../../../data/programPlan";
+import WeekConfirmationDetails from "../WeekConfirmationDetails";
+
+function ListOfWeeksWithDetail(props) {
+  const { userData } = props;
+  const { t } = useTranslation();
+
+  const baseQuestionNames = [
+    "tooSick",
+    "tooSickNumberOfDays",
+    "couldNotAcceptWork",
+    "didYouLook",
+    "refuseWork",
+  ];
+  const uiOnlyQuestionNames = ["schoolOrTraining", "workOrEarn"];
+  const puaOnlyQuestionNames = [
+    "otherBenefits",
+    "workOrEarn",
+    "recentDisaster",
+    "disasterChoice",
+  ];
+
+  // TODO(kalvin): also used in RCCertificationPage, refactor to common place
+  function getQuestionKey(transKey, weekProgramPlan) {
+    if (weekProgramPlan === programPlan.uiPartTime) {
+      return "retrocerts-certification.questions.ui-part-time." + transKey;
+    }
+    if (weekProgramPlan === programPlan.uiFullTime) {
+      return "retrocerts-certification.questions.ui-full-time." + transKey;
+    }
+    if (weekProgramPlan === programPlan.puaFullTime) {
+      return "retrocerts-certification.questions.pua-full-time." + transKey;
+    }
+  }
+
+  function getSubmittedAnswer(questionName, weekData) {
+    const answer = weekData[questionName];
+    if (answer === undefined) {
+      // The only case where the answer should be undefined is for
+      // the "number of days you were sick" question, and then only
+      // if the answer to "were you sick" was yes
+      if (questionName !== "tooSickNumberOfDays" || weekData.tooSick) {
+        // TODO turn this into an error we log
+        console.log("missing answer", questionName, weekData);
+      }
+      return "N/A";
+    }
+
+    // disasterChoice answers are stored in the format "choice-3 TEXT OF Q..."
+    if (questionName === "disasterChoice") {
+      return answer.substring(answer.indexOf(" ") + 1);
```

Comment:
For this answer, the question is: what should happen if the user changes language? Should the answer text also change language? By using the key, the text will change, although we store the localized text in the database. FWIW, I think this is fine and I would keep it this way, but maybe you have a different opinion. Also something to consider for the staff access page.

Reply:
Interesting... the way it's implemented now, for this answer and other free text answers, it uses the localized text that was originally submitted by the user. So if the user changes language, these answers won't change (but the "Yes" "No" true false answers will change). I don't see any advantage of using the key for this answer, do you?

Reply:
Oh, you're right! I misread the code. Yes, this will use the answer the user picked. That seems fine to me.

Reply:
Oh, that's an interesting point! I think I agree... cc @nguyen-tina @mkhandekar should we always show the English text in the staff admin app even when the user selected an option in Spanish?

Reply:
discussed with Tina, the answer is we should show whatever language is selected by the toggle since that’s the expectation when you select a language (not the language the user filled out the question in)